### ISSUE TITLE
Need option 'Remove Array Bounds Checks'

### ISSUE THREAD

**Issue by WaynePhillipsEA** [twinbasic/twinbasic#1166]
VB6 offers various optimization options for final build EXE/DLLs etc.  `Remove Array Bounds Checks` is one of them.

This is an important option to offer as part of the tB compiler, because some VB6 projects require it ON in order to work correctly (or shall we say identically to the VB6 compiler).

Note that in VB6 these options DO NOT effect the debugger environment; only final builds.  We may want to consider allowing this also for code running under the debugger, optionally of course.

**Comment by dmrvb** [comment]
At some future stage it would be great if these options could be set on a module by basis, or even better a routine by routine basis.
In particular, removing array bounds checks can give a huge speed improvement on some mathematical code, but its not something that can be safely done on all code.

**Comment by WaynePhillipsEA** [comment]
Yes that's a good point.  I guess we could apply it via attributes.  Something like `[ Optimize(RemoveArrayBounds) ]`.  Alternatively we add an `Option Optimize ...` syntax if we feel the characteristics are too strong for attributes.

**Comment by rexxitall** [comment]
I think a optimize syntax for the whole project and a by routine attribute would be great. In the User Interface i would like to have those checks. Users cmes on strange ideas and speed is there never a issue. In the workling code of a 3d model i wont have them for speed reasons. And yes sure we want also see in the debugger whts going on. Otherwise you drive blind in the final build :)

**Comment by dmrvb** [comment]
Yes, the difference between behaviour in the IDE and final EXE has caught me out seriously on a couple of occassions and lead to some very hard to track down bugs.

**Comment by WaynePhillipsEA** [comment]
Already implemented.
